Documentation/test-targets-design.md

```diff
@@ -0,0 +1,87 @@
+# Test Target Design
+
+This document describes the design of the BuildTools test targets. For documentation on their usage, see [Test Targets](test-targets-usage.md).
+
+The primary entry points are the [`Test` target](https://github.com/dotnet/buildtools/blob/73375848b73a586a8ac439a1263b605901414ad8/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L336) in Microsoft.DotNet.Build.Tasks and the [`CloudBuild` target](https://github.com/dotnet/buildtools/blob/73375848b73a586a8ac439a1263b605901414ad8/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets#L79) in Microsoft.DotNet.Build.CloudTestTasks, which are each described below.
```

Comment:
is referencing a specific version of the file intentional here?  (i.e. to keep the line # accurate or something)